hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/services/DefinitionsService.java

```diff
@@ -654,9 +654,30 @@ void updateActionPlugin(String actionPlugin, Map<String, String> defaultProperti
 
     ActionDefinition getActionDefinition(String tenantId, String actionPlugin, String actionId) throws Exception;
 
+    /**
+     * Register a local DefinitionsListener.
+     * A local listener is only invoked in the node that performs the definitions operation.
+     *
+     * @param listener the Listener to add
+     * @param eventType specify a type to filter when the listener will be invoked
+     * @param eventTypes optional types to filter when the listener will be invoked
+     */
     void registerListener(DefinitionsListener listener, Type eventType, Type... eventTypes);
 
     /**
+     * Register a local or distributed Definitions listener.
+     * A local listener is only invoked in the node that performs the definitions operation.
+     * A distributed listener is invoked in all the nodes of the cluster.
+     *
+     * @param listener the listener to add
+     * @param distributed true indicates listener is distributed
+     *                    false indicates listener is local
+     * @param eventType specify a type to filter when the listener will be invoked
+     * @param eventTypes optional types to filter when the listener will be invoked
+     */
+    void registerListener(DefinitionsListener listener, boolean distributed, Type eventType, Type... eventTypes);
```

Comment:
Maybe for clarity we should offer an explicit registerDistributedListener() as opposed to using a flag.